external-crates/move/crates/move-binary-format/src/compatibility.rs

```diff
@@ -208,25 +394,20 @@ impl Compatibility {
         // friend list. But for simplicity, we decided to go to the more restrictive form now and
         // we may revisit this in the future.
         for (name, old_func) in &old_module.functions {
+            // if we don't find a function in the new module
             let Some(new_func) = new_module.functions.get(name) else {
-                if old_func.visibility == Visibility::Friend {
-                    friend_linking = false;
-                } else if old_func.visibility != Visibility::Private {
-                    datatype_and_function_linking = false;
-                } else if old_func.is_entry && self.check_private_entry_linking {
-                    // This must be a private entry function. So set the link breakage if we're
-                    // checking for that.
-                    entry_linking = false;
-                }
+                context.function_missing(old_func, self.check_private_entry_linking);
```

Comment:
Having to rely on the `self.check_private_entry_linking` feels a little strange here, would it make more sense to split this out into further cases? e.g.

```rust
if old.func_visibility == Visibility::Friend {
    context.friend_function_missing(old_func);
} else if old_func.visibility != Visibility::Private {
    context.public_function_missing(old_func);
} else if old_func.is_entry && self.check_private_entry_linking {
    context.entry_function_missing(old_func);
}
```

Reply:
> I could pull out the functions to missing_friend, missing_private, missing_entry if thats what you are thinking.

Yeah, this is what I had in mind -- generally keeping the separation between the configuration of what matters, and the tracking of what went wrong into two distinct phases.